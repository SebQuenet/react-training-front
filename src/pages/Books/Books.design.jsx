// @ts-nocheck
import React from 'react';

import styles from './Books.module.scss';

const myPage = () => (
  <div className={styles.container} />
);

export default myPage;
