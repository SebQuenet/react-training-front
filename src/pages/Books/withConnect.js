import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchBooks, getAllBooksSelector } from 'domains/books';

const mapStateToProps = store => ({
  books: getAllBooksSelector(store),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchBooks,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps);
