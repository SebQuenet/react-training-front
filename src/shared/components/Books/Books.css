.Books__descriptionCards {
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: center;
  text-align: left;
  margin-top: 32px;
  margin-bottom: 8px;
  padding: 0 32px;
  display: flex;
  overflow-x: auto;
}
