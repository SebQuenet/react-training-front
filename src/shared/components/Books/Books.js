// @flow

import React from 'react';
import { connect } from 'react-redux';
import Book from './Book/Book';

import './Books.css';

type PropTypes = {
  books: Array<Object>,
};

/**
 * Books component
 */
const Books = ({ books }: PropTypes) => {
  return (
    <ul className="Books__descriptionCards">
      {books && books.map((book, index) => <Book key={index} book={book} />)}
    </ul>
  );
};

/**
 * @private
 * @param {Object} state
 * @returns {{}}
 */
const mapStateToProps = (state: string) => ({
  ...state,
});

export default connect(mapStateToProps)(Books);
