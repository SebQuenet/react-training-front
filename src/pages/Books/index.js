import { compose, lifecycle } from 'recompose';

import Books from './Books.design';
import withConnect from './withConnect';

export default compose(
  withConnect,
  lifecycle({
    componentDidMount() {

    },
  }),
)(Books);
