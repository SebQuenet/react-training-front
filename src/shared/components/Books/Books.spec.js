import React from 'react';
import renderer from 'react-test-renderer';
import Books from './Books';

jest.mock('react-redux', () => ({ connect: () => c => c }));
jest.mock('./Book/Book', () => 'mock-book-component');

const defaultProps = {
  books: [
    {
      url: 'https://anapioficeandfire.com/api/books/1',
      name: 'A Game of Thrones',
      isbn: '978-0553103540',
      authors: ['George R. R. Martin'],
      numberOfPages: 694,
      publisher: 'Bantam Books',
      country: 'United States',
      mediaType: 'Hardcover',
      released: '1996-08-01T00:00:00',
      characters: [
        'https://anapioficeandfire.com/api/characters/2',
        'https://anapioficeandfire.com/api/characters/12',
        'https://anapioficeandfire.com/api/characters/13',
        'https://anapioficeandfire.com/api/characters/16',
        'https://anapioficeandfire.com/api/characters/20',
        'https://anapioficeandfire.com/api/characters/27',
        'https://anapioficeandfire.com/api/characters/31',
      ],
      povCharacters: [
        'https://anapioficeandfire.com/api/characters/148',
        'https://anapioficeandfire.com/api/characters/208',
        'https://anapioficeandfire.com/api/characters/232',
        'https://anapioficeandfire.com/api/characters/339',
      ],
    },
    {
      url: 'https://anapioficeandfire.com/api/books/2',
      name: 'A Clash of Kings',
      isbn: '978-0553108033',
      authors: ['George R. R. Martin'],
      numberOfPages: 768,
      publisher: 'Bantam Books',
      country: 'United States',
      mediaType: 'Hardback',
      released: '1999-02-02T00:00:00',
      characters: [
        'https://anapioficeandfire.com/api/characters/2',
        'https://anapioficeandfire.com/api/characters/12',
        'https://anapioficeandfire.com/api/characters/13',
        'https://anapioficeandfire.com/api/characters/16',
      ],
      povCharacters: [
        'https://anapioficeandfire.com/api/characters/148',
        'https://anapioficeandfire.com/api/characters/208',
        'https://anapioficeandfire.com/api/characters/232',
        'https://anapioficeandfire.com/api/characters/583',
      ],
    },
    {
      url: 'https://anapioficeandfire.com/api/books/3',
      name: 'A Storm of Swords',
      isbn: '978-0553106633',
      authors: ['George R. R. Martin'],
      numberOfPages: 992,
      publisher: 'Bantam Books',
      country: 'United States',
      mediaType: 'Hardcover',
      released: '2000-10-31T00:00:00',
      characters: [
        'https://anapioficeandfire.com/api/characters/2',
        'https://anapioficeandfire.com/api/characters/3',
        'https://anapioficeandfire.com/api/characters/7',
        'https://anapioficeandfire.com/api/characters/12',
        'https://anapioficeandfire.com/api/characters/13',
        'https://anapioficeandfire.com/api/characters/16',
        'https://anapioficeandfire.com/api/characters/20',
      ],
      povCharacters: [
        'https://anapioficeandfire.com/api/characters/148',
        'https://anapioficeandfire.com/api/characters/208',
        'https://anapioficeandfire.com/api/characters/232',
        'https://anapioficeandfire.com/api/characters/529',
      ],
    },
  ],
};

describe('Books component', () => {
  it('should render correctly', () => {
    // given
    const component = renderer.create(<Books {...defaultProps} />);
    // when
    const tree = component.toJSON();
    // then
    expect(tree).toMatchSnapshot();
  });
});
