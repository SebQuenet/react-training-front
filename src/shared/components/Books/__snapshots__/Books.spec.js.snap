// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books component should render correctly 1`] = `
<ul
  className="Books__descriptionCards"
>
  <mock-book-component
    book={
      Object {
        "authors": Array [
          "George R. R. Martin",
        ],
        "characters": Array [
          "https://anapioficeandfire.com/api/characters/2",
          "https://anapioficeandfire.com/api/characters/12",
          "https://anapioficeandfire.com/api/characters/13",
          "https://anapioficeandfire.com/api/characters/16",
          "https://anapioficeandfire.com/api/characters/20",
          "https://anapioficeandfire.com/api/characters/27",
          "https://anapioficeandfire.com/api/characters/31",
        ],
        "country": "United States",
        "isbn": "978-0553103540",
        "mediaType": "Hardcover",
        "name": "A Game of Thrones",
        "numberOfPages": 694,
        "povCharacters": Array [
          "https://anapioficeandfire.com/api/characters/148",
          "https://anapioficeandfire.com/api/characters/208",
          "https://anapioficeandfire.com/api/characters/232",
          "https://anapioficeandfire.com/api/characters/339",
        ],
        "publisher": "Bantam Books",
        "released": "1996-08-01T00:00:00",
        "url": "https://anapioficeandfire.com/api/books/1",
      }
    }
  />
  <mock-book-component
    book={
      Object {
        "authors": Array [
          "George R. R. Martin",
        ],
        "characters": Array [
          "https://anapioficeandfire.com/api/characters/2",
          "https://anapioficeandfire.com/api/characters/12",
          "https://anapioficeandfire.com/api/characters/13",
          "https://anapioficeandfire.com/api/characters/16",
        ],
        "country": "United States",
        "isbn": "978-0553108033",
        "mediaType": "Hardback",
        "name": "A Clash of Kings",
        "numberOfPages": 768,
        "povCharacters": Array [
          "https://anapioficeandfire.com/api/characters/148",
          "https://anapioficeandfire.com/api/characters/208",
          "https://anapioficeandfire.com/api/characters/232",
          "https://anapioficeandfire.com/api/characters/583",
        ],
        "publisher": "Bantam Books",
        "released": "1999-02-02T00:00:00",
        "url": "https://anapioficeandfire.com/api/books/2",
      }
    }
  />
  <mock-book-component
    book={
      Object {
        "authors": Array [
          "George R. R. Martin",
        ],
        "characters": Array [
          "https://anapioficeandfire.com/api/characters/2",
          "https://anapioficeandfire.com/api/characters/3",
          "https://anapioficeandfire.com/api/characters/7",
          "https://anapioficeandfire.com/api/characters/12",
          "https://anapioficeandfire.com/api/characters/13",
          "https://anapioficeandfire.com/api/characters/16",
          "https://anapioficeandfire.com/api/characters/20",
        ],
        "country": "United States",
        "isbn": "978-0553106633",
        "mediaType": "Hardcover",
        "name": "A Storm of Swords",
        "numberOfPages": 992,
        "povCharacters": Array [
          "https://anapioficeandfire.com/api/characters/148",
          "https://anapioficeandfire.com/api/characters/208",
          "https://anapioficeandfire.com/api/characters/232",
          "https://anapioficeandfire.com/api/characters/529",
        ],
        "publisher": "Bantam Books",
        "released": "2000-10-31T00:00:00",
        "url": "https://anapioficeandfire.com/api/books/3",
      }
    }
  />
</ul>
`;
